File: vpon_plugin_poc/android/src/main/kotlin/io/flutter/plugins/vponmobileads/VponFlutterAd.java
package io.flutter.plugins.vponmobileads;

abstract class VponFlutterAd {

    protected final int adId;

    VponFlutterAd(int adId) {
        this.adId = adId;
    }

    abstract void load();
    abstract void dispose();

    abstract static class VponFlutterOverlayAd extends VponFlutterAd {

        abstract void show();

        VponFlutterOverlayAd(int adId) {
            super(adId);
        }
    }
}

File: vpon_plugin_poc/android/src/main/kotlin/io/flutter/plugins/vponmobileads/VponFlutterAd.kt
package io.flutter.plugins.vponmobileads;

import androidx.annotation.Nullable;

import io.flutter.plugin.platform.PlatformView;

abstract class VponFlutterAd {

    protected final int adId;

    VponFlutterAd(int adId) {
        this.adId = adId;
    }

    abstract void load();
    abstract void dispose();

    @Nullable
    PlatformView getPlatformView() {
        return null;
    }

    abstract static class VponFlutterOverlayAd extends VponFlutterAd {

        abstract void show();

        VponFlutterOverlayAd(int adId) {
            super(adId);
        }
    }
}
